### ISSUE TITLE
Add sprite option to dbg_button (Imgui ImageButton())

### ISSUE THREAD

**Issue by blaiddx64** [YoYoGames/GameMaker-Bugs#6087]
### Is your feature request related to a problem?

The Imgui have the ImageButton() function that allows developers to create buttons like these below:
![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/103244351/90be48e6-89a6-4555-b57e-f2181b84a3b0)

It would be useful for a lot of things such as list different things like items, enemies, etc that can be clicked and spawn or anything else. 

### Describe the solution you'd like

Add the option to set a sprite instead of text on dbg_button:
```gml
dbg_button(spr_item, ref, 32, 32);
```

Or add a new function:
```gml
dbg_image_button(spr_item, ref, 32, 32);
```

**Comment by rwkay** [comment]
Fixed in 2024.8 - added new function `dbg_sprite_button(label,dbgRef,dbgRefSprite,dbgRefSpriteIndex,[width],[height],[xoffset],[yoffset],[widthSprite],[heightSprite]...)`

Creates a button within the current debug section, the given method will be called when the button is pressed; it is not split into 2 columns but shown in a single column. The sprite is specified with the given reference, the width and height of the button specified in pixels (width and height of the sprite is used if not specified). Bits of the sprite can be displayed by specifying the x and y offset along with width and height (if none given then the whole sprite is used). NOTE: if no section is declared a "Default" section will be created.

**Comment by YYBartT** [comment]
Documented the new function `dbg_sprite_button()`. All changes are currently in `develop.bart`.

**Comment by cameron-home** [comment]
Verified fixed in IDE v2024.800.0.593  Runtime v2024.800.0.617 using the attached project.

[OptionalSpriteButtonParameters.zip](https://github.com/user-attachments/files/16283745/OptionalSpriteButtonParameters.zip)
